webrecorder/browsertrix
# Allow users to rate page in QA review

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1494

Should be rebased if https://github.com/webrecorder/browsertrix-cloud/pull/1576 is merged first

### Changes

Adds button group to QA review page for approving, rejecting, and adding comments to a page.

### Manual testing

1. Log in and go to "Archived Items"
2. Click a crawl and add `/review/screenshots` to the end of the URL. Verify page approval buttons are shown between "Next" and "Previous" page buttons.
3. Click thumbs up and thumbs down. Verify approval saves as expected.
4. Click comment icon to add an comment. Verify comment saves as expected.
5. Click comment icon again and click "Comments". Verify saved comment is displayed with name and date.
6. Click trash icon next to comment. Verify comment is deleted.

Changes from mockups:
- Active comment button is kept a white background to make styling the active and inactive button styles simpler. I did give it a college try using background gradients (see older commits) but the animating the transition between inactive rounded corners and active no border looked odd.
- Add/remove multiple comments, per [discord thread](https://discord.com/channels/895426029194207262/1214733913214423050/1214733916649300008).
- We don't have a user guide page for comment data usage (yet?) so I added some copy, definitely open to wording suggestions.

### Screenshots

| Page | Image/video |
| ---- | ----------- |
|      |             |
| QA Review | Default: <img width="204" alt="Screenshot 2024-03-05 at 4 44 05 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/9c5a0b9a-ce30-4e2a-9067-b4aaa770286f"><br/>Hover: <img width="211" alt="Screenshot 2024-03-05 at 4 44 17 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/a6f65384-ec28-4758-82d8-48e63d10ed57"><img width="203" alt="Screenshot 2024-03-05 at 4 44 21 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/23594d0a-49e2-4f0f-8635-945bf4f9741e"><br/>Active:<img width="206" alt="Screenshot 2024-03-05 at 4 44 27 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/bcb51107-a00f-4e6f-8aa9-ff60da2411e0"><img width="203" alt="Screenshot 2024-03-05 at 4 44 32 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/86ad53a7-c024-4c33-900c-5badff7cdee4"><img width="200" alt="Screenshot 2024-03-05 at 4 44 55 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/6a2ac1f7-1dc2-49e0-842a-97d1f0adcd02"> |
QA Review - Comments dialog | No comments: <img width="535" alt="Screenshot 2024-03-05 at 4 44 42 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/cf150c81-8c67-424d-8bab-80cb07ecadcb"><br/>With comments: <img width="520" alt="Screenshot 2024-03-05 at 4 45 20 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/1a486469-39f2-48df-8891-ed382c8bc113"> |
